apps/guardian-ui/src/admin/FederationAdmin.tsx

```diff
@@ -34,15 +34,18 @@ export const FederationAdmin: React.FC = () => {
 
   useEffect(() => {
     if (!inviteCode) return;
-    api.config(inviteCode).then(setConfig).catch(console.error);
+    api.config().then(setConfig).catch(console.error);
   }, [inviteCode, api]);
 
+  // TODO: FIXME to use better loading state
+  if (!config || !status) return <Text>{'Loading...'}</Text>;
```

Comment:
If `FederationInfoCard` allows for `config` to be `undefined`, we don't need this early return at all. Probably better to delegate each component to handle its own loading state instead of one huge one.